backend/app/routers/datasets.py

```diff
@@ -498,6 +506,7 @@ async def add_folder(
     folder_in: FolderIn,
     user=Depends(get_current_user),
     db: MongoClient = Depends(dependencies.get_db),
+    allow: bool = Depends(DatasetAuthorization("editor")),
```

Comment:
there is also a role between viewer and editor that is "uploader". I think a few of these could use the more relaxed uploader role instead if they involve only additive things - maybe add_folder, save_file, get_dataset_extract?

Reply:
I put uploader for these.